### DISCUSSION TITLE
Disk named s3 cannot be accessed

### DISCUSSION THREAD

**Discussion by onursahindur** [spatie/laravel-medialibrary#3328] [Q&A]
Hello,
I started getting an error as `Disk named s3 cannot be accessed.

My s3 config is as follows;

```
     's3' => [
            'driver' => 's3',
            'key' => env('AWS_ACCESS_KEY_ID'),
            'secret' => env('AWS_SECRET_ACCESS_KEY'),
            'region' => env('AWS_DEFAULT_REGION'),
            'bucket' => env('AWS_BUCKET'),
            'url' => env('AWS_URL'),
            'endpoint' => env('AWS_ENDPOINT'),
            'use_path_style_endpoint' => env('AWS_USE_PATH_STYLE_ENDPOINT', false),
            'throw' => false,
        ],
```
        
Does anyone know why this occurs? Thanks
Thanks.

**Comment by MattWiseParking** [comment]
Hi

I am also getting this issue.  I have read a couple of articles and some suggest the following:

in the .env file wrap all the variable data in double quotes, for example:  

`AWS_DEFAULT_REGION="eu-west-2"`

however this didnt help me.  another article suggested:

specify the AWS_ENDPOINT in the .env file.  a list of the end points are found in this aws document:
[https://docs.aws.amazon.com/general/latest/gr/s3.html](url)

again this didnt help me.

I also then double checked the permissions against the s3 bucket and appear to be ok.  i will continue to identify what the issue is

**Comment by MattWiseParking** [comment]
the issue is that you need to remove the aws variables from the .env file, because it automatically populates them:

`AWS_ENDPOINT`
`AWS_URL`

**Comment by onursahindur** [comment]
Hello, 
Thanks for all responses. 

Actually the problem was because of we first used DigitalOcean Spaces, then reverted to AWS S3.

In order to work with Spaces we had to put an `ACL` extra header with `public-read` value in `config/media-library.php`.
After we removed that header, everything started to work as expected again.

So if anyone gets an issue like this in the future, just keep in mind that.

Check the commit here:
<img width="274" alt="Screenshot 2023-08-03 at 23 13 56" src="https://github.com/spatie/laravel-medialibrary/assets/5830807/4628edbd-d8e7-42e5-bb8b-9e7b26855909">

Cheers.
